uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/Menu/Parser.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use CRM_HRCore_Menu_Navigation as Navigation;
+
+class CRM_HRCore_Helper_Menu_Parser {
+
+  /**
+   * Builds the navigation menu tree and returns in a format expected by Civi
+   * in order to generate the navigation menu.
+   *
+   * @param Navigation $navigationMenu
+   *
+   * @return array
+   *  Sample return format:
+   *  [
+   *    'attributes' =>
+   *      [
+   *        'label' => 'Components',
+   *         'url' => 'civicrm/bla',
+   *         'permission' => 'sample permission',
+   *         'operator' => 'AND',
+   *      ],
+   *   ],
+   */
+  public static function getNavigationTree(Navigation $navigationMenu) {
```

Comment:
How about avoiding static methods and making this a container service? I know it doesn't have external calls, but it might in the future. What do you think?

Reply:
Good idea.